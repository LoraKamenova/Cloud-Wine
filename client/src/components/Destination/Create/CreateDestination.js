import './CreateDestination.css'
import FormHeader from "../../Shared/FormHeader/FormHeader";
import InputError from "../../Shared/InputError/InputError";

import * as destinationService from "../../../services/destinationService";

import {useHistory} from "react-router-dom"
import {useState} from "react";

const CreateDestination = () => {
    const history = useHistory();
    const [errorMessage, setErrorMessage] = useState('');

    const onCreateDestinationSubmitHandler = (e) => {
        e.preventDefault();

        const {
            name,
            description,
            address,
            services,
            logoUrl,
            imageUrl1,
            imageUrl2,
            mapUrl
        } = e.target;

        destinationService.create(
            name.value,
            description.value,
            address.value,
            services.value,
            logoUrl.value,
            imageUrl1.value,
            imageUrl2.value,
            mapUrl.value)
            .then(() => {
                history.push('/destination/all')
            })
    };

    const onChangeDescriptionHandler = (e) => {
        if (e.target.value.length < 100) {
            setErrorMessage('Описанието трябва да е минимум 100 символа');
        } else {
            setErrorMessage('');
        }
    };

    return (
        <section className="form-wrapper add-destination-form-wrapper">
            <FormHeader title="Нова дестинация"/>

            <form className="form-container" onSubmit={onCreateDestinationSubmitHandler}>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="name">Име:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="name"
                           name="name"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="address">Адрес:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="address"
                           name="address"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="description">Описание:</label>
                    </div>
                    <textarea className="form-field-input textarea"
                              rows="11"
                              id="description"
                              name="description"
                              required
                              onBlur={onChangeDescriptionHandler}/>
                    <InputError>{errorMessage}</InputError>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="services">Услуги:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="services"
                           name="services"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="logoUrl">URL лого:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="logoUrl"
                           name="logoUrl"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="imageUrl1">URL снимка 1:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="imageUrl1"
                           name="imageUrl1"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="imageUrl2">URL снимка 2:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="imageUrl2"
                           name="imageUrl2"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="mapUrl">URL карта:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="mapUrl"
                           name="mapUrl"
                           required/>
                </div>

                <div className="form-button-wrapper">
                    <input className="form-button" type="submit" value="ИЗПРАТИ"/>
                </div>

            </form>
        </section>
    );
};

export default CreateDestination;
