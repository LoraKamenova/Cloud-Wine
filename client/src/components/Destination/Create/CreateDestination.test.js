import { render } from '@testing-library/react';
import CreateDestination from './CreateDestination';

describe('CreateDestination Component', () => {

    it("CreateDestination Component snapshot", () => {
        const { asFragment } = render(<CreateDestination />);
        expect(asFragment()).toMatchSnapshot();
    });

});
