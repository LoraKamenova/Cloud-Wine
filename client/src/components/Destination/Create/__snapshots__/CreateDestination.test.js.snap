// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDestination Component CreateDestination Component snapshot 1`] = `
<DocumentFragment>
  <section
    class="form-wrapper add-destination-form-wrapper"
  >
    <div
      class="form-header-wrapper"
    >
      <div
        class="form-header-heading-container"
      >
        <h1
          class="form-header-heading"
        >
          Нова дестинация
        </h1>
      </div>
      <div
        class="form-header-image-container"
      >
        <img
          alt="vine"
          class="form-header-image"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1615712293/React%20Project/grape_border_ogd8gq.png"
        />
      </div>
      <h1
        class="form-header-heading-hidden"
      >
        Нова дестинация
      </h1>
    </div>
    <form
      class="form-container"
    >
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="name"
          >
            Име:
          </label>
        </div>
        <input
          class="form-field-input"
          id="name"
          name="name"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="address"
          >
            Адрес:
          </label>
        </div>
        <input
          class="form-field-input"
          id="address"
          name="address"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="latitude"
          >
            Географска ширина:
          </label>
        </div>
        <input
          class="form-field-input"
          id="latitude"
          name="latitude"
          placeholder="пример: 24.1751"
          required=""
          step="0.0001"
          type="number"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="longitude"
          >
            Географска дължина:
          </label>
        </div>
        <input
          class="form-field-input"
          id="longitude"
          name="longitude"
          placeholder="пример: 42.2658"
          required=""
          step="0.0001"
          type="number"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="description"
          >
            Описание:
          </label>
        </div>
        <textarea
          class="form-field-input textarea"
          id="description"
          name="description"
          required=""
          rows="11"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="services"
          >
            Услуги:
          </label>
        </div>
        <input
          class="form-field-input"
          id="services"
          name="services"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="logoUrl"
          >
            URL лого:
          </label>
        </div>
        <input
          class="form-field-input"
          id="logoUrl"
          name="logoUrl"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="imageUrl1"
          >
            URL снимка 1:
          </label>
        </div>
        <input
          class="form-field-input"
          id="imageUrl1"
          name="imageUrl1"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="imageUrl2"
          >
            URL снимка 2:
          </label>
        </div>
        <input
          class="form-field-input"
          id="imageUrl2"
          name="imageUrl2"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="mapUrl"
          >
            URL карта:
          </label>
        </div>
        <input
          class="form-field-input"
          id="mapUrl"
          name="mapUrl"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-button-wrapper"
      >
        <input
          class="form-button"
          type="submit"
          value="ИЗПРАТИ"
        />
      </div>
    </form>
  </section>
</DocumentFragment>
`;
